### PR TITLE
[DX-1070] TT-11258 Update tyk-sync.md

### PR DESCRIPTION
**Pull request by carolwkc**
Update Tyk Sync regarding compatibility with 5.3 OAS API

### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

[DX-1070](https://tyktech.atlassian.net/browse/DX-1070)

---

### Preview Link

<br>

[Preview](https://deploy-preview-4140--tyk-docs.netlify.app/docs/nightly/tyk-sync/)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

[DX-1070]: https://tyktech.atlassian.net/browse/DX-1070?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tyk-docs/content/tyk-sync.md [hunk 1]
```diff
@@ -41,9 +41,7 @@ Tyk Sync works with the Open Source *Tyk Gateway* and *Tyk Dashboard* installati
...
-
-In Tyk Dashboard 5.2.2 we added a new [configuration option]({{< ref "tyk-dashboard/configuration#allow_unsafe_oas" >}}) `allow_unsafe_oas` which permits the modification of Tyk OAS APIs via the Tyk Classic API endpoints. In Tyk Sync 1.4.1 we added a new flag `--allow-unsafe-oas` to work with a Dashboard configured with this option enabled. This is not recommended action due to the risk of inconsistent behaviour and potential for breaking changes while Tyk OAS is in Early Access, but is provided for early adopters and will be deprecated later.
+Tyk Sync supports you to work with [Tyk OAS APIs]({{< ref "getting-started/key-concepts/high-level-concepts" >}}). Tyk Sync is currently using classic Dashboard API so the use of [allow-unsafe-oas]({{< ref "tyk-dashboard/configuration#allow_unsafe_oas" >}}) configuration in Dashboard is required if you want to use Tyk Sync to migrate APIs and Policies. Please also the  flag `--allow-unsafe-oas` in Tyk Sync if you are working with OAS APIs. OAS API Category is currently not supported in Tyk Sync.
```
Comment (lghiur) [inline tyk-docs/content/tyk-sync.md]: I would add a note that saying that this option will suffer changes in the future, and must be carefully used. Also I would wait for @andyo-tyk to review this,

Comment (carolwkc) [reply tyk-docs/content/tyk-sync.md]: Updated to mention this flag will be deprecated in future.

File: tyk-docs/content/tyk-sync.md [hunk 2]
```diff
@@ -13,11 +13,7 @@ Tyk Sync is a command-line tool and Go library for synchronising API definitions
...
 
-{{< note success >}}
-**Note**  
-
-Tyk Sync works with APIs and Policies. It does not work with Keys. See [Move Keys between environments]({{< ref "advanced-configuration/manage-multiple-environments/move-keys-between-environments" >}}) for details.
-{{< /note >}}
+Tyk Sync does not work with Keys. See [Move Keys between environments]({{< ref "advanced-configuration/manage-multiple-environments/move-keys-between-environments" >}}) for details.
```
Comment (dcs3spp) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
Tyk Sync does not work with keys. Please consult [move keys between environments]({{< ref "advanced-configuration/manage-multiple-environments/move-keys-between-environments" >}}) for further details.
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
PR LGTM Carol, I have made small suggestion and after peer review approved by @lghiur I will then merge and release.

File: tyk-docs/content/tyk-sync.md [hunk 3]
```diff
@@ -41,9 +37,9 @@ Tyk Sync works with the Open Source *Tyk Gateway* and *Tyk Dashboard* installati
 {{< note success >}}
 **Note**  
 
-Tyk Sync supports you to work with [Tyk OAS APIs]({{< ref "getting-started/key-concepts/high-level-concepts" >}}), however as Tyk OAS is currently in [Early Access]({{< ref "frequently-asked-questions/using-early-access-features" >}}) we do not recommend their use in production environments.
+Tyk Sync supports you to work with [Tyk OAS APIs]({{< ref "getting-started/key-concepts/high-level-concepts" >}}). Tyk Sync is currently using classic Dashboard API so the use of [allow-unsafe-oas]({{< ref "tyk-dashboard/configuration#allow_unsafe_oas" >}}) configuration in Dashboard, and the flag `--allow-unsafe-oas` in Tyk Sync is required if you want to use Tyk Sync to migrate OAS APIs. OAS API Category is currently not supported in Tyk Sync.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
Tyk Sync supports both [Tyk OAS APIs]({{< ref "getting-started/key-concepts/high-level-concepts" >}}) and [Tyk Classic APIs]({{< ref "[ggetting-started/key-concepts/what-is-an-api-definition/#api-definition-types" >}}) when working with Tyk Dashboard, however at this time you must set the [allow-unsafe-oas]({{< ref "tyk-dashboard/configuration#allow_unsafe_oas" >}}) configuration in Dashboard, and the flag `--allow-unsafe-oas` when invoking Tyk Sync if you want to use Tyk Sync to migrate Tyk OAS APIs.

API Category is not currently supported in Tyk Sync for Tyk OAS APIs.
```

Comment (andyo-tyk) [reply tyk-docs/content/tyk-sync.md]: Question: here we're telling users to set `allow_unsafe_oas` to work with Tyk OAS APIs... but this is a Dashboard config, not relevant to OSS Gateway... so does Sync actually support Tyk OAS APIs for OSS users? (re: the table lower down the page that gives a green tick for OSS)

File: tyk-docs/content/tyk-sync.md [hunk 4]
```diff
@@ -41,9 +37,9 @@ Tyk Sync works with the Open Source *Tyk Gateway* and *Tyk Dashboard* installati
...
 
-In Tyk Dashboard 5.2.2 we added a new [configuration option]({{< ref "tyk-dashboard/configuration#allow_unsafe_oas" >}}) `allow_unsafe_oas` which permits the modification of Tyk OAS APIs via the Tyk Classic API endpoints. In Tyk Sync 1.4.1 we added a new flag `--allow-unsafe-oas` to work with a Dashboard configured with this option enabled. This is not recommended action due to the risk of inconsistent behaviour and potential for breaking changes while Tyk OAS is in Early Access, but is provided for early adopters and will be deprecated later.
+`--allow-unsafe-oas` is an option provided for early adopters and will be deprecated later. Tyk Sync will be updated in a future release to use Dashboard API for OAS to bring you the full OAS API experience.
```
Comment (andyo-tyk) [inline tyk-docs/content/tyk-sync.md]:
```suggestion
`--allow-unsafe-oas` is a flag that is required for Tyk Sync to work with Tyk OAS APIs. This is a temporary measure provided for early adopters and will be deprecated later when Tyk Sync will be updated in a future release to bring you the full Tyk OAS API experience. 
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Suggested tweaks to wording - and a question re: Tyk OAS + OSS Gateway

File: tyk-docs/content/tyk-sync.md (hunk 3 above)
Comment (carolwkc) [reply tyk-docs/content/tyk-sync.md]: @andyo-tyk Sync use classic API endpoints to create/update OAS APIs when working with OSS Gateway. The `allow_unsafe_oas` config might be required for OSS Gateway too, can you confirm if it was implemented in Gateway?

**Review by dcs3spp — APPROVED** [review]

**Comment by dcs3spp** [comment]
Thanks @caroltyk I have approved. @andyo-tyk @lghiur, if the changes updated by @caroltyk for peer review are ok can they be approved ready for merge and release?

**Comment by dcs3spp** [comment]
@andyo-tyk @lghiur 

@caroltyk Has made requested peer review changes, would you be able to approve if ok?

**Review by andyo-tyk — APPROVED** [review]
Fine by me, thanks.

File: tyk-docs/content/tyk-sync.md (hunk 1 above)
Comment (dcs3spp) [reply tyk-docs/content/tyk-sync.md]: > I would add a note that saying that this option will suffer changes in the future, and must be carefully used. Also I would wait for @andyo-tyk to review this,

@andyo-tyk @caroltyk @lghiur See there is a label on PR for 5.3.0 so double checking. This should not yet be released yet, yes?

Comment (lghiur) [reply tyk-docs/content/tyk-sync.md]: @dcs3spp as 5.3.0 is not out yet, no

Comment (dcs3spp) [reply tyk-docs/content/tyk-sync.md]: ok thanks @lghiur @lghiur When 5.3.0 released can the update of this PR also be released to 5.2 etc. or is it only intended for 5.3.0+

**Comment by dcs3spp** [comment]
@andyo-tyk @carol @lghiur I will merge and then release later for 5.3.0

**Pull request merged**